### PR TITLE
Fix: Add return objects for requests and correct return types in proto for event service

### PR DESCRIPTION
**Pull request by ZixMai**
1.  Added return objects for requests.
2.  Corrected return types in proto for event service.
3.  Moved big queries to constants and made query more readable.

### CODE REVIEW

**Review by MaxPlays35 — APPROVED** [review]

File: event_service/repository/event_repository_impl.py [hunk 1]
```diff
@@ -362,8 +373,9 @@ async def update_event(self, event: Event) -> None:
         await self._db_client.db.event.update(
             where={"id": event.id}, data=event.to_dict()
         )
+        return await self.get_event_by_event_id(event.id)
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: No need use this https://github.com/prisma/prisma/discussions/17604

File: event_service/repository/event_repository_impl.py [hunk 2]
```diff
@@ -372,33 +384,49 @@ async def delete_event_by_id(self, event_id: str) -> None:
...
         prisma.errors.PrismaError
             Catch all for every exception raised by Prisma Client Python.
 
         """
+        event = await self.get_event_by_event_id(event_id)
         await self._db_client.db.event.update_many(
             where={"id": event_id, "deleted_at": None},
             data={"deleted_at": datetime.now()},
         )
+        return event
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: No need to return here

File: event_service/repository/event_repository_impl.py [hunk 3]
```diff
@@ -337,8 +342,9 @@ async def create_event(self, event: Event) -> None:
         await self._db_client.db.event.create(
             data=event.to_dict(exclude=["created_at", "deleted_at"])
         )
+        return await self.get_event_by_event_id(event.id)
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: No need use this https://github.com/prisma/prisma/discussions/17604

File: event_service/repository/event_repository_impl.py [hunk 4]
```diff
@@ -372,33 +384,49 @@ async def delete_event_by_id(self, event_id: str) -> None:
...
 
         """
+        events = await self.get_events_by_author_id(
+            author_id, page_number=1, items_per_page=-1
+        )
         await self._db_client.db.event.update_many(
             where={"author_id": author_id, "deleted_at": None},
             data={"deleted_at": datetime.now()},
         )
+        return events
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: No need to return here

File: event_service/repository/event_repository_interface.py [hunk 5]
```diff
@@ -210,7 +220,7 @@ async def update_event(self, event: Event) -> None:
         pass
 
     @abstractmethod
-    async def delete_event_by_id(self, event_id: str) -> None:
+    async def delete_event_by_id(self, event_id: str) -> Event:
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_interface.py]: No need to return here

File: event_service/repository/event_repository_interface.py [hunk 6]
```diff
@@ -228,7 +243,7 @@ async def delete_event_by_id(self, event_id: str) -> None:
         pass
 
     @abstractmethod
-    async def delete_events_by_author_id(self, author_id: str) -> None:
+    async def delete_events_by_author_id(self, author_id: str) -> List[Event]:
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_interface.py]: No need to return here

File: event_service/repository/mock_event_repository.py [hunk 7]
```diff
@@ -336,17 +347,23 @@ async def update_event(self, event: Event) -> None:
                 self._events[index] = event
             else:
                 raise ValueNotFoundError("Events authors must be same")
+            return event
         except StopIteration:
             raise ValueNotFoundError("Event not found")
 
-    async def delete_event_by_id(self, event_id: str) -> None:
+    async def delete_event_by_id(self, event_id: str) -> Event:
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: No need to return here

File: event_service/repository/mock_event_repository.py [hunk 8]
```diff
@@ -360,18 +377,26 @@ async def delete_event_by_id(self, event_id: str) -> None:
...
             )
+            deleted_event = await self.get_event_by_event_id(event_id)
             self._events[index].deleted_at = datetime.now()
+            return deleted_event
+
         except StopIteration:
             raise ValueNotFoundError("Event not found")
 
-    async def delete_events_by_author_id(self, author_id: str) -> None:
+    async def delete_events_by_author_id(self, author_id: str) -> List[Event]:
```
Comment (Nhsdkk) [inline event_service/repository/mock_event_repository.py]: No need to return here

File: event_service/src/event_service_impl.py [hunk 9]
```diff
@@ -233,11 +220,11 @@ async def create_event(
             raise PermissionDeniedError("Permission denied")
         await self._event_repository.create_event(event=event)
         context.set_code(grpc.StatusCode.OK)
-        return Empty()
+        return request.event
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: Wrong return object

File: event_service/src/event_service_impl.py [hunk 10]
```diff
@@ -267,11 +254,11 @@ async def update_event(
         event = Event.from_grpc_event(request.event)
         await self._event_repository.update_event(event=event)
         context.set_code(grpc.StatusCode.OK)
-        return Empty()
+        return request.event
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: Wrong return object

File: event_service/src/event_service_impl.py [hunk 11]
```diff
@@ -267,11 +254,11 @@ async def update_event(
...
 
     async def delete_event_by_id(
         self, request: proto.DeleteEventByIdRequest, context: grpc.ServicerContext
-    ) -> Empty:
+    ) -> proto.GrpcEvent:
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: No need to return here

File: event_service/src/event_service_impl.py [hunk 12]
```diff
@@ -301,13 +288,13 @@ async def delete_event_by_id(
...
         context.set_code(grpc.StatusCode.OK)
-        return Empty()
+        return event.to_grpc_event()
 
     async def delete_events_by_author_id(
         self,
         request: proto.DeleteEventsByAuthorIdRequest,
         context: grpc.ServicerContext,
-    ) -> Empty:
+    ) -> proto.ListOfEvents:
```
Comment (Nhsdkk) [inline event_service/src/event_service_impl.py]: No need to return here

File: shared/proto/event_service/event_service.proto [hunk 13]
```diff
@@ -2,18 +2,17 @@ syntax = "proto3";
...
-  rpc update_event(EventRequest) returns (google.protobuf.Empty){}
-  rpc delete_event_by_id(DeleteEventByIdRequest) returns (google.protobuf.Empty){}
-  rpc delete_events_by_author_id(DeleteEventsByAuthorIdRequest) returns (google.protobuf.Empty){}
+  rpc get_event_by_event_id(EventRequestByEventId) returns (GrpcEvent){}
+  rpc get_events_by_events_ids(EventsRequestByEventsIds) returns (ListOfEvents){}
+  rpc get_events_by_author_id(EventsRequestByAuthorId) returns (ListOfEvents){}
+  rpc get_all_events(GetAllEventsRequest) returns (ListOfEvents){}
+  rpc create_event(EventRequest) returns (GrpcEvent){}
+  rpc update_event(EventRequest) returns (GrpcEvent){}
+  rpc delete_event_by_id(DeleteEventByIdRequest) returns (GrpcEvent){}
```
Comment (Nhsdkk) [inline shared/proto/event_service/event_service.proto]: No need to return here

File: shared/proto/event_service/event_service.proto [hunk 14]
```diff
@@ -2,18 +2,17 @@ syntax = "proto3";
...
+  rpc delete_events_by_author_id(DeleteEventsByAuthorIdRequest) returns (ListOfEvents){}
```
Comment (Nhsdkk) [inline shared/proto/event_service/event_service.proto]: No need to return here

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

File: event_service/repository/event_repository_impl.py [hunk 15]
```diff
@@ -339,10 +362,9 @@ async def create_event(self, event: Event) -> Event:
             raise WrongIntervalError(
                 "Request failed. Can't create event with wrong time interval."
             )
-        await self._db_client.db.event.create(
+        return await self._db_client.db.event.create(
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Wrong return type

File: event_service/repository/event_repository_impl.py [hunk 16]
```diff
@@ -370,12 +392,11 @@ async def update_event(self, event: Event) -> Event:
             raise WrongIntervalError(
                 "Request failed. Can't create event with wrong time interval."
             )
-        await self._db_client.db.event.update(
+        return await self._db_client.db.event.update(
```
Comment (Nhsdkk) [inline event_service/repository/event_repository_impl.py]: Wrong return type

**Review by Nhsdkk — CHANGES_REQUESTED** [review]
Make changes

**Review by Nhsdkk — APPROVED** [review]
LGTM

**Pull request merged**